bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/resource/ResourceProvider.java

```diff
@@ -32,4 +34,18 @@ public interface ResourceLoader {
 	 * @return Returns a list of objects that are of the target type
 	 */
 	public <T extends EObject> List<T> lookupElementOfType(final EClass targetType);
+	
+	/**
+	 * Looks up an ECore element with the given class type
+	 * @param id  Id of the objects that the lookup should return
+	 * @return Returns the object with the given id
+	 */
+	public EObject lookupElementWithId(String id);
+	
+	/**
+	 * Finds an element that satisfies the given condition
+	 * @param condition Condition the element should satisfy
+	 * @return Returns the first element found that satisfies the given condition
+	 */
+	public Entity findElement(Predicate<Entity> condition);
```

Comment:
Are the elements returned by `lookupElementWithId` and `findElements` always Entites, sometimes EObjects? The signature differs between both functions and also between the interface and its implementations in the provider classes